Clarification on Span Creation API (#322)
open-telemetry/opentelemetry-specification: specification/api-tracing.md

active `Span` by default, but this functionality MAY be offered additionally
as a separate operation.

The API MUST require the caller to provide:

- The operation name.

> - The parent Span or parent Span context, and whether the new `Span` should be a

Reviewer: This somewhat depends on the language's interaction with the Context. E.g. in Go the Context must be passed (and returned), which may or may not contain the parent span.

Also, this list is under REQUIRED, whereas a parent is not a required parameter. Why not have a single list (joined with L259+) and just marking the arguments as optional or required?

Reviewer 2: Split into "The API MUST ALLOW the caller to provide:" and "The API MUST REQUIRE the caller to provide:" sections, the parent span/span context/"I want a root span" option being in the former.

Reviewer 3: Couldn't we use the `Link` parameters to specify the parent context? Otherwise, do we need `Link` parameters?

A `Span` is said to have a _remote parent_ if it is the child of a `Span`
created in another process. Each propagators' deserialization must set
`IsRemote` to true on a parent `SpanContext` so `Span` creation knows if the
parent is remote.

The API MUST allow users to provide the following properties:

- [`SpanKind`](#spankind), default to `SpanKind.Internal` if not specified.
- `Attribute`s - similar API with [Span::SetAttributes](#set-attributes). These
  attributes will be used to make a sampling decision as noted in [sampling
  description](sdk-tracing.md#sampling).
- `Link`s - see API definition [here](#add-links).

> - `Start timestamp`, default to current time. This argument SHOULD only be set

Reviewer 2: Best add a note here that the API must still only be called for in-band spans, even if providing a timestamp.

Author: What API must be used for out-of-band Spans than?

Reviewer 2: There is no API for that, but you may directly create the SDK-level readable Span objects yourself and feed them to some exporter you construct yourself. This OTEP by @bogdandrutu removed support for out of band reporting because it greatly increased complexity of the API: https://github.com/open-telemetry/oteps/blob/master/text/0007-no-out-of-band-reporting.md

Author: let's handle out of band spans in a separate PR.

  when span creation time has already passed. If API is called at a moment of

>   a Span logical start, API user must rely on SDK to take take the current time.

Reviewer 2: ```suggestion
  a Span logical start, API user MUST not explicitly set this argument.
```
